From 6540157fe389b3dc46e9a1306370acd79d40c183 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Wed, 7 Jun 2023 13:41:15 +0300
Subject: [PATCH] [llvm-dlltool] Ignore the --temp-prefix option

llvm-dlltool tolerates unknown options as long as they are plain
flags, but if given the parameter value as a separate argument,
e.g. "--temp-prefix foo", it fails to ignore it.

Differential Revision: https://reviews.llvm.org/D152361
---
 llvm/lib/ToolDrivers/llvm-dlltool/Options.td  | 3 +++
 llvm/test/tools/llvm-dlltool/ignored-opts.def | 8 ++++++++
 2 files changed, 11 insertions(+)
 create mode 100644 llvm/test/tools/llvm-dlltool/ignored-opts.def

diff --git a/llvm/lib/ToolDrivers/llvm-dlltool/Options.td b/llvm/lib/ToolDrivers/llvm-dlltool/Options.td
index e78182ab8130..6da5dc8f5814 100644
--- a/llvm/lib/ToolDrivers/llvm-dlltool/Options.td
+++ b/llvm/lib/ToolDrivers/llvm-dlltool/Options.td
@@ -24,3 +24,6 @@ def S_alias: JoinedOrSeparate<["--"], "as">, Alias<S>;
 
 def f: JoinedOrSeparate<["-"], "f">, HelpText<"Assembler Flags">;
 def f_alias: JoinedOrSeparate<["--"], "as-flags">, Alias<f>;
+
+def t: JoinedOrSeparate<["-"], "t">, HelpText<"Prefix for temporary files (ignored)">;
+def t_alias: JoinedOrSeparate<["--"], "temp-prefix">, Alias<t>;
diff --git a/llvm/test/tools/llvm-dlltool/ignored-opts.def b/llvm/test/tools/llvm-dlltool/ignored-opts.def
new file mode 100644
index 000000000000..6cc05e376d31
--- /dev/null
+++ b/llvm/test/tools/llvm-dlltool/ignored-opts.def
@@ -0,0 +1,8 @@
+; RUN: llvm-dlltool -m i386 -d %s -l %t.a --temp-prefix foo
+; RUN: llvm-dlltool -m i386 -d %s -l %t.a --temp-prefix=foo
+; RUN: llvm-dlltool -m i386 -d %s -l %t.a -t foo
+; RUN: llvm-dlltool -m i386 -d %s -l %t.a -tfoo
+
+LIBRARY test.dll
+EXPORTS
+TestFunction
-- 
2.41.0.162.gfafddb0af9-goog

